packages/web3-react/src/helpers/regex.ts

```diff
@@ -0,0 +1,5 @@
+const pattern = new RegExp(
```

Comment:
I guess it is not critical at all, but maybe it is better to use this regex for urls:
https://gist.github.com/dperini/729294
You can see some tests here: https://mathiasbynens.be/demo/url-regex

Reply:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#extracting_sub-domain_name_from_url), it is better to use `URL` constructor to test urls for validity, e.g.
```
function isValidUrl(url) {
  try {
    new URL(url)
    return true
  } catch {
    return false
  }
}
```